mlir/test/Transform/AIRDependency/reshape_matmul_nd.mlir

```diff
@@ -0,0 +1,106 @@
+//===- matmul_nd.mlir ------------------------------------------*- MLIR -*-===//
+//
+// Copyright (C) 2022, Xilinx Inc. All rights reserved.
+// Copyright (C) 2022, Advanced Micro Devices, Inc. All rights reserved.
+// SPDX-License-Identifier: MIT
+//
+//===----------------------------------------------------------------------===//
+
+// RUN: air-opt %s -air-dependency | FileCheck %s
+
+module attributes {torch.debug_module_name = "mmult"}  {
+  func.func @forward(%arg0: memref<64x64xi32>, %arg1: memref<64x16x4xi32>, %arg2: memref<?x?xi32>) {
+    %c0_i32 = arith.constant 0 : i32
+    %c2 = arith.constant 2 : index
+    %c0 = arith.constant 0 : index
+    %c4 = arith.constant 4 : index
+    %c16 = arith.constant 16 : index
+    %c32 = arith.constant 32 : index
+    %c64 = arith.constant 64 : index
+    %c1 = arith.constant 1 : index
+    %0 = memref.alloc() : memref<64x64xi32>
+    // CHECK: = air.execute
+    // CHECK: air.execute_terminator
+    linalg.fill ins(%c0_i32 : i32) outs(%0 : memref<64x64xi32>)
+    // CHECK: = air.execute
+    %1 = memref.cast %arg2 : memref<?x?xi32> to memref<64x64xi32>
```

Comment:
Is this `memref.cast` the only code pattern that you want to test for? If so, can we reduce the test size to focus on this op?

Reply:
Actually, let me just remove memref.cast. We are testing for memref.alloc and memref.stores associated with the memref.reshape op.